internal/e2e/main_test.go

```diff
@@ -251,6 +250,88 @@ func SetMinioLDAPIdentityProvider(td TestData) {
 	}
 }
 
+// LogLevelHandler configures a [slog.Handler] with a log level derived from the environment.
+type LogLevelHandler struct {
+	slog.Handler
+}
+
+func (src *LogLevelHandler) GetLevel() slog.Level {
+	l := slog.Level(100)
+	if os.Getenv("VERBOSE") == "1" {
+		l = slog.LevelDebug
+	}
+	return l
+}
+
+// Part of the [LogLevelHandler] implementation of [slog.Handler]
+func (src *LogLevelHandler) Enabled(ctx context.Context, lvl slog.Level) bool {
+	return lvl >= src.GetLevel()
+}
+
+// Used by [LogLevelHandler.Handle] to prevent error logs from being emitted.
+type DoNotLog struct{}
+
+func (e *DoNotLog) Error() string {
+	return "do not log"
+}
+
+// Part of the [LogLevelHandler] implementation of [slog.Handler]
+// NOTE: [logr.Logger] will log error messages regardless of verbosity - which is why this method is required
+// NOTE: [DoNotLog] is returned (vs. nil) - otherwise the log-line prefix will be emitted.
+func (src *LogLevelHandler) Handle(ctx context.Context, r slog.Record) error {
+	if r.Level >= src.GetLevel() {
+		return src.Handler.Handle(ctx, r)
+	}
+	return &DoNotLog{}
+}
+
+// Part of the [LogLevelHandler] implementation of [slog.Handler]
+func (src *LogLevelHandler) WithAttrs(attrs []slog.Attr) slog.Handler {
+	return &LogLevelHandler{Handler: src.Handler.WithAttrs(attrs)}
+}
+
+// Part of the [LogLevelHandler] implementation of [slog.Handler]
+func (src *LogLevelHandler) WithGroup(name string) slog.Handler {
+	return &LogLevelHandler{Handler: src.Handler.WithGroup(name)}
+}
+
+// Returns a [slogt.Option] wrapping a [slogt.Bridge]'s [slog.Handler] in a [LogLevelHandler]
+func WithLogLevelHandler() slogt.Option {
+	return func(b *slogt.Bridge) {
+		b.Handler = &LogLevelHandler{Handler: b.Handler}
+	}
+}
+
+// LogCollector wraps a [slog.Handler] and collects emitted records
+type LogCollector struct {
```

Comment:
rather than hold an inner `slogt.Bridge` - make `LogCollector` compose `slog.Handler` so that it can inherit the composed methods.